### PR TITLE
Dbatiste/visual diff

### PR DESCRIPTION
**Pull request by dbatiste**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,3 +17,5 @@ env:
   global:
   - SAUCE_USERNAME: Desire2Learn
```
Comment (dlockhart) [inline .travis.yml]: It'd be nice to get the comments back in here so we can tell which variable is which. I'm assuming that first one is `# SAUCE_ACCESS_KEY`.

Comment (dbatiste) [reply .travis.yml]: Agreed - added comments for those.  :)

**Comment by dbatiste** [comment]
Yeah agreed.  There's certainly some potential to re-use some of the setup/init.

On the topic of `getRect`: the diff image is only generated if the dimensions of the current/golden match.  The consumer would have to visually compare the current/golden screenshots without the aid of the diff image.  That's perhaps ok.  Alternatively, there may be ways we can ensure we almost always get a  diff image: the consumer can specify a hard-coded clip region; or we can modify `getRect` to take a larger screenshot to build in some tolerance for subtle changes in dimensions of the component.  They can also choose to screenshot the entire page, but that seems excessive.

I am still a little unsettled on where to store the golden screenshots.  Currently it's S3 when running in CI.  It is runnable locally as well.  S3 has the advantage of being able to host and view the results, including the generated diff image.  GitHub has the advantage of tracking history, identifying conflicts, etc.  Currently golden update logic also identifies and removes orphaned goldens... a step that would also be needed if we went the GitHub route. @svanherk and I were chatting a bit more earlier and perhaps we could do both to get the best of both worlds.  For now we can run the workflow with S3 for a bit and we can decide if we want to change.  

It probably be good for us to chat briefly about how the remote goldens are currently updated.  I can demo that.  I'll set something up.

**Comment by dlockhart** [comment]
Sounds good Dave -- yeah let's do a demo and chat, I'm super interested in the inner workings here!

**Pull request merged**